ee-9/source-transform/ee/src/main/java/org/jboss/as/ee/concurrent/resource/definition/ManagedExecutorDefinitionInjectionSource.java

```diff
@@ -0,0 +1,215 @@
+ *
+ * @author emmartins
+ */
+public class ManagedExecutorDefinitionInjectionSource extends ResourceDefinitionInjectionSource {
+
+    public static final String CONTEXT_PROP = "context";
+    public static final String HUNG_TASK_THRESHOLD_PROP = "hungTaskThreshold";
+    public static final String MAX_ASYNC_PROP = "maxAsync";
+
+    private static final String REQUEST_CONTROLLER_CAPABILITY_NAME = "org.wildfly.request-controller";
+
+    private String contextServiceRef;
+    private long hungTaskThreshold;
+    private int maxAsync = (ProcessorInfo.availableProcessors() * 2);
+    private int hungTaskTerminationPeriod = 0;
+    private boolean longRunningTasks = false;
+    private long keepAliveTime = 60000;
+    private TimeUnit keepAliveTimeUnit = TimeUnit.MILLISECONDS;
+    private long threadLifeTime = 0L;
+    private int queueLength = Integer.MAX_VALUE;
+    private AbstractManagedExecutorService.RejectPolicy rejectPolicy = AbstractManagedExecutorService.RejectPolicy.ABORT;
+    private int threadPriority = Thread.NORM_PRIORITY;
+
+    public ManagedExecutorDefinitionInjectionSource(final String jndiName) {
+        super(jndiName);
+    }
+
+    public void getResourceValue(final ResolutionContext context, final ServiceBuilder<?> serviceBuilder, final DeploymentPhaseContext phaseContext, final Injector<ManagedReferenceFactory> injector) throws DeploymentUnitProcessingException {
+        final String resourceName = uniqueName(context);
+        final String resourceJndiName = "java:jboss/ee/concurrency/definition/managedExecutor/"+resourceName;
+        final CapabilityServiceSupport capabilityServiceSupport = phaseContext.getDeploymentUnit().getAttachment(org.jboss.as.server.deployment.Attachments.CAPABILITY_SERVICE_SUPPORT);
+
+        try {
+            // install the resource service
+            final ServiceName resourceServiceName = ManagedExecutorServiceResourceDefinition.CAPABILITY.getCapabilityServiceName(resourceName);
+            final ServiceBuilder resourceServiceBuilder = phaseContext.getServiceTarget().addService(resourceServiceName);
+            final Supplier<ManagedExecutorHungTasksPeriodicTerminationService> hungTasksPeriodicTerminationService = resourceServiceBuilder.requires(ConcurrentServiceNames.HUNG_TASK_PERIODIC_TERMINATION_SERVICE_NAME);
+            final ManagedExecutorServiceService resourceService = new ManagedExecutorServiceService(resourceName, resourceJndiName, hungTaskThreshold, hungTaskTerminationPeriod, longRunningTasks, maxAsync, maxAsync, keepAliveTime, keepAliveTimeUnit, threadLifeTime, queueLength, rejectPolicy, threadPriority, hungTasksPeriodicTerminationService);
+            resourceServiceBuilder.setInstance(resourceService);
+            final String contextServiceRef = this.contextServiceRef == null || this.contextServiceRef.isEmpty() ? "java:comp/DefaultContextService" : this.contextServiceRef;
```

Comment:
This could be right as I'm not real familiar with how this works in WildFly. However, the `ManagedExecutorDefinition.context()` default seems like it should be used. It does seem to be "java:comp/DefaultContextService", but it just seems a bit fragile to hard-code it.

Reply:
Not sure I am following, you would prefer to grab the JNDI address for the platform's default ContextService from somewhere else?

Reply:
Sorry. I just mean the `ManagedExecutorDefinition` annotation has defaults and it just feels a little cleaner to use the default rather than the hard-coded version. The hard-coded string does match the default so it's not critical, but if the default were to ever change we could see potential issues.

Reply:
Changed to the const we had in the deployer responsible for binding it.